File: 121_BestTimeToBuyAndSellStock.py
# 22.76%
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        diffPrices = map(lambda x, y: x - y, prices[1:], prices[:-1])
        maxCurr = 0
        maxAll = 0
        for diffPrice in diffPrices:
            maxCurr = max(0, maxCurr + diffPrice)
            maxAll = max(maxCurr, maxAll)
        return maxAll

File: 121_BestTimeToBuyAndSellStock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxAll = 0;
		int maxCurr = 0;
		for(int i = 1; i < prices.size(); i++){
			maxCurr = max(0, maxCurr + prices[i] - prices[i - 1]);
			maxAll = max(maxCurr, maxAll);
		}
		return maxAll;
    }
};